SquareLine/gen.py

```diff
@@ -0,0 +1,231 @@
+from genericpath import isdir
+import sys
+import os
+import json
+import shutil
+from colorama import Fore
+from zipfile import ZipFile
+
+# Directory of all boards
+BOARDS_DIR = "boards/"
+# Other board files, which are same for all boards (there can be used placeholders)
+COMMON_DIR = "common/"
+# JSON file which describing the board (must be in each board directory)
+BOARD_GENFILE = "manifest.json"
+# Files, which are specific for each board. Will be copied from boards/BOARD/ directory
+BOARD_SPECIFIC_FILES = {"sdkconfig.defaults", "main/idf_component.yml"}
+# Generated output directory
+OUT_DIR = "espressif"
+# SquareLine output directory
+SQUARELINE_OUTDIR = "__ui_project_name__"
+# Template of SLB file
+SLB_FILE = {
+    "version":"",
+    "group":"Espressif",
+    "title": "",
+    "keywords": "Espressif",
+    "width": 0,
+    "height": 0,
+    "width_min": 0,
+    "height_min": 0,
+    "width_max": 0,
+    "height_max": 0,
+    "offset_x": 0,
+    "offset_y": 0,
+    "rotation": 0,
+    "color_depth": "",
+    "lvgl_export_path": "",
+    "lvgl_include_path": "lvgl.h",
+    "supported_lvgl_version": "8.0+",
+    "pattern_match_files": "./CMakeLists.txt",
+    "language":"C",
+    "ui_export_path":"./main/ui",
+    "url":"https://github.com/espressif/esp-bsp",
+    "short_description": "",
+    "long_description": ""
+}
+
+global placeholders
+
+# Print green text
+def print_ok(str):
+    print(Fore.GREEN + str + Fore.WHITE)
+
+# Print red error
+def print_error(str):
+    print(Fore.RED + str + Fore.WHITE)
+
+def remove_folder(path):
+    for filename in os.listdir(path):
+        f = os.path.join(path, filename)
+        if os.path.isdir(f):
+            remove_folder(path + "/" + filename)
+        else:
+            os.remove(path + "/" + filename)
+    os.rmdir(path)
+
+class SafeDict(dict):
+    def __missing__(self, key):
+        return '{' + key + '}'
+
+def replace_placeholders(file):
+    global placeholders
+    if os.path.exists(file):
+        f = open(file, "r")
+        str = f.read()
+        f.close()
```

Comment:
```suggestion
        with open(file, "r") as f:
            str = f.read()
```

(similar for other file operations)

Another note, `str` is a built-in type name in Python. Please try picking a different variable name to avoid overwriting the built-in type.